homeassistant/components/vizio/config_flow.py

```diff
@@ -165,9 +175,11 @@ def __init__(self) -> None:
         """Import a config entry from configuration.yaml."""
         # Check if new config entry matches any existing config entries
         for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if entry.data[CONF_HOST] == import_config[CONF_HOST] and entry.data[
-                CONF_NAME
-            ] == import_config.get(CONF_NAME):
+            if _host_is_same(
+                entry.data[CONF_HOST], import_config[CONF_HOST]
+            ) and _name_is_same(
```

Comment:
Comparing on name feels very fragile ? A user renames their TV and it's seen as a new device? Do we not have access to a serial number ?

Reply:
Only have access to the serial number once HA can talk to the API. The intent here is to allow config users to change the `volume_step` option without having to create an entirely new entry. I can update this to make the name updatable as well since before I have the serial number, the host is the implicit unique_id